src/ert/_c_wrappers/enkf/_config_content_as_dict.py

```diff
@@ -92,8 +92,10 @@ def config_content_as_dict(
             if key in SINGLE_OCCURRENCE_SINGLE_ARG_KEYS:
                 content_dict[key] = item.getValue()
             elif key in SINGLE_OCCURRENCE_MULTI_ARG_KEYS:
+                if key not in content_dict:
+                    content_dict[key] = []
                 for node in item:
-                    content_dict[key] = list(node)
+                    content_dict[key].append(list(node))
```

Comment:
so i have a question about this.

single occurence multi arg means we can have the keyword several times in the config file, but we only care about the last time it appears? because if that is so, then i think this should still be an assignment rather than an append, but with another list around it:
```
            elif key in SINGLE_OCCURRENCE_MULTI_ARG_KEYS:
                for node in item:
                    content_dict[key] = [list(node)]
```

then for something like

```
SCHEDULE_PREDICTION_FILE some-template PARAMETER:input/param INIT_FILES:input/int
```

you get the following for the schedule key in the config content:

```
[["some-template", "PARAMETER:input/param", "INIT_FILES:input/int"]]
```

and then this works with the schedule_file_list that is passed to ensemble config.

then again, this is single occurence, and if we say we only care about the last one and discard the others from the content dict, then really we should stick to the original version of this function, and rather change the schedule file list that ensemble config expect to be not a list of (several) schedule files, but a list of strings representing a single schedule file, and update the ensemble config code a bit. i'm sorry, i'm too lazy to reproduce the code changes in this silly github web interface, if what i mean is not clear, we can maybe just have a look at the code together tomorrow, and get this done and merged.

Reply:
good point :+1:

Reply:
great, thank you!